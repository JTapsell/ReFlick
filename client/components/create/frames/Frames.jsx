import React from 'react'
import { connect } from 'react-redux'
import Frame from './Frame'

import Pen from '../draw/Pen'

class Frames extends React.Component {
  renderFrames (frames, dispatch) {
    return (
      frames.map(frame => {
        if (frame.name !== 'None') { // Speical case to avoid rendering frame 0 (dropdown frame)
          return <Frame key={frame.id} frame={frame.name} dispatch={dispatch} update={() => this.props.update}/>
        }
      })
    )
  }

  render () {
    const { frames } = this.props
    return (
      <div className="frames-pen">
        <div className="frames">
          <h4> Frames </h4>
          <div className="framesContent">
            {this.renderFrames(frames, this.props.dispatch)}
          </div>
          <br/>
        </div>

        <Pen fillOn={this.props.fillOn} fillOff={this.props.fillOff}/>
      </div>
    )
  }
}

function mapStateToProps (state) {
  return {
    frames: state.framesReducer
  }
}

export default connect(mapStateToProps)(Frames)
