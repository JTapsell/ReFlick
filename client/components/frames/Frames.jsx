import React from 'react'
import {connect} from 'react-redux'
import Frame from './Frame'

class Frames extends React.Component {
  renderFrames (frames, dispatch) {
    return (
      frames.map(frame =>
        <Frame key={frame.id} frame={frame.name} dispatch={dispatch} />
      )
    )
  }

  render () {
    const {frames} = this.props
    return (
      <div className = "right">
        <h2> Frames </h2>
        {this.renderFrames(frames, this.props.dispatch)}
      </div>
    )
  }
}

function mapStateToProps (state) {
  return {
    frames: state.framesReducer
  }
}

export default connect(mapStateToProps)(Frames)
