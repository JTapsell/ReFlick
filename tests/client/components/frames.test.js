import Frames from '../../../client/components/frames/Frames'
import { shallow } from 'enzyme'
import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'

const mockStore = configureMockStore()
const store = mockStore({
  frames: [
    { id: 1, name: 1 },
    { id: 2, name: 2 },
    { id: 3, name: 3 },
    { id: 4, name: 4 }
  ]
})

test('test runner is working', () => {
  expect(true).toBeTruthy()
})
let wrapper
beforeEach(() => {
  wrapper = shallow(<Provider store={store}><Frames /></Provider>)
})
describe('<Frames is running', () => {
  it('Renders correct', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it(('renders store'), () => {
    expect(wrapper).toEqual({})
  })
})
