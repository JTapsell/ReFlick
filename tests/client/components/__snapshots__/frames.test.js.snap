// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Frames is running Renders correct 1`] = `ShallowWrapper {}`;
